framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -49,6 +49,8 @@ MeshDiagnosticsGenerator::validParams()
       "check_for_watertight_nodesets",
       chk_option,
       "whether to check for external nodes that are not assigned to any nodeset");
+  params.addParam<std::vector<BoundaryID>>(
```

Comment:
```suggestion
  params.addParam<std::vector<BoundaryName>>(
```
let s do names. ids, especially for boundaries, are really error-prone

Reply:
So I was looking into this and I think I ran into a problem. For boundary ids there's a map from element to boundary id but there's no equivalent for element to boundary names. There are functions for getting boundary names from the ID but I feel like that defeats the purpose?

Reply:
the user interface asks for names.

then shortly after in the constructor or the setup we take the vector of names and convert it to a vector of ids